show/bgp_quagga_v4.py

```diff
@@ -1,6 +1,71 @@
 import click
+import json
+import re
+from natsort import natsorted
+from tabulate import tabulate
+from natsort import natsorted
+from tabulate import tabulate
 from show.main import *
 
+def parse_bgp_summary(summ):
+    """turn output of show ip bgp summary to a list"""
+    ls = summ.splitlines()
+    bgpinfo = []
+    # Read until the table header
+    n = len(ls)
+    li = 0
+    while li < n:
+        l = ls[li]
+        if l.startswith('Neighbor        '): break
+        if l.startswith('No IPv'):  # eg. No IPv6 neighbor is configured
+            return bgpinfo
+        if l.endswith('> exit'):  # last command in the lines
+            return bgpinfo
+        li += 1
+    # Read and store the table header
+    if li >= n:
+        raise ValueError('No table header found')
+    hl = ls[li]
+    li += 1
+    ht = re.split('\s+', hl.rstrip())
+    hn = len(ht)
+    # Read rows in the table
+    while li < n:
+        l = ls[li]
+        li += 1
+        if l == '': break
+        # Handle line wrap
+        # ref: bgp_show_summary in https://github.com/Azure/sonic-quagga/blob/debian/0.99.24.1/bgpd/bgp_vty.c
+        if ' ' not in l:
+            # Read next line
+            if li >= n:
+                raise ValueError('Unexpected line wrap')
+            l += ls[li]
+            li += 1
+        # Note: State/PfxRcd field may be 'Idle (Admin)'
+        lt = re.split('\s+', l.rstrip(), maxsplit=hn - 1)
+        if len(lt) != hn:
+            raise ValueError('Unexpected row in the table')
+        dic = dict(zip(ht, lt))
+        bgpinfo.append(dic)
+    return bgpinfo
+
+
+def run_command_save(command, display_cmd=False):
+    """run a command and save the output"""
+    if display_cmd:
+        click.echo(click.style("Command: ", fg='cyan') + click.style(command, fg='green'))
+    try:
+        proc = subprocess.Popen(command,
+                                stdout=subprocess.PIPE,
+                                shell=True,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        result = stdout.rstrip('\n')
+    except OSError, e:
+        raise OSError("Error running command")
```

Comment:
>OSError [](start = 14, length = 7)

raise the same type of exception, no benefit added, and lose the inner exception info.